src/ReactiveMarbles.Locator/ServiceLocatorMixins.cs

```diff
@@ -18,122 +15,66 @@ public static class ServiceLocatorMixins
         /// <summary>
         /// Adds the service.
         /// </summary>
-        /// <typeparam name="T">The type used for registration.</typeparam>
-        /// <param name="serviceLocator">The service locator.</param>
-        /// <param name="instanceFactory">The instance factory.</param>
-        /// <param name="contract">The contract.</param>
-        public static void AddService<T>(this IEditServices serviceLocator, Func<object?> instanceFactory, string? contract = null) => serviceLocator.AddService(instanceFactory, typeof(T), contract);
-
-        /// <summary>
-        /// Adds the service.
-        /// </summary>
-        /// <typeparam name="TInterface">The type of the interface.</typeparam>
+        /// <typeparam name="TContract">The type of the interface.</typeparam>
         /// <typeparam name="TConcrete">The type of the concrete.</typeparam>
         /// <param name="serviceLocator">The service locator.</param>
         /// <param name="contract">The contract.</param>
-        public static void AddService<TInterface, TConcrete>(this IEditServices serviceLocator, string? contract = null)
-            where TConcrete : new() =>
-            serviceLocator.AddService(() => new TConcrete(), typeof(TInterface), contract);
+        public static void AddService<TContract, TConcrete>(this IEditServices serviceLocator, string contract)
+            where TConcrete : TContract, new() =>
+            serviceLocator.AddService<TContract>(() => new TConcrete(), contract);
 
         /// <summary>
         /// Adds the singleton.
         /// </summary>
-        /// <typeparam name="T">The type used for registration.</typeparam>
+        /// <typeparam name="TContract">The type used for registration.</typeparam>
         /// <param name="serviceLocator">The service locator.</param>
         /// <param name="instance">The instance.</param>
         /// <param name="contract">The contract.</param>
-        public static void AddSingleton<T>(this IEditServices serviceLocator, object instance, string? contract = null) => serviceLocator.AddService(() => instance, typeof(T), contract);
+        public static void AddSingleton<TContract>(this IEditServices serviceLocator, object instance, string contract) =>
```

Comment:
Can we avoid the whole "object" thing here?

Reply:
These are left over and will probably go away in the next iteration.  I can remove the whole class now that most of the methods exist on the interface.